src/languages/r.js

```diff
@@ -2,76 +2,148 @@
+            'is.numeric is.object is.pairlist is.raw is.recursive is.single' +
+            'is.symbol lazyLoadDBfetch length lgamma list log max min' +
+            'missing Mod names nargs nzchar oldClass on.exit pos.to.env' +
+            'proc.time prod quote range Re rep retracemem return round' +
+            'seq_along seq_len seq.int sign signif sin sinh sinpi sqrt' +
+            'standardGeneric substitute sum switch tan tanh tanpi tracemem' +
+            'trigamma trunc unclass untracemem UseMethod xtfrm',
         },
         relevance: 0
       },
-      {
-        // hex value
-        className: 'number',
-        begin: "0[xX][0-9a-fA-F]+[Li]?\\b",
-        relevance: 0
-      },
-      {
-        // explicit integer
-        className: 'number',
-        begin: "\\d+(?:[eE][+\\-]?\\d*)?L\\b",
-        relevance: 0
-      },
-      {
-        // number with trailing decimal
-        className: 'number',
-        begin: "\\d+\\.(?!\\d)(?:i\\b)?",
-        relevance: 0
-      },
-      {
-        // number
-        className: 'number',
-        begin: "\\d+(?:\\.\\d*)?(?:[eE][+\\-]?\\d*)?i?\\b",
-        relevance: 0
-      },
-      {
-        // number with leading decimal
-        className: 'number',
-        begin: "\\.\\d+(?:[eE][+\\-]?\\d*)?i?\\b",
-        relevance: 0
-      },
+
+      // Roxygen comments
+      hljs.COMMENT(
+        /#'/,
+        /$/,
+        {
+          contains: [
+            {
+              // Handle `@examples` separately to cause all subsequent code
+              // until the next `@`-tag on its own line to be kept as-is,
+              // preventing highlighting. This code is example R code, so nested
+              // doctags shouldn’t be treated as such. See
+              // `test/markup/r/roxygen.txt` for an example.
+              className: 'doctag',
+              begin: '@examples',
+              starts: {
+                contains: [
+                  { begin: /\n/ },
+                  {
+                    begin: /#'\s*(?=@[a-zA-Z]+)/,
+                    endsParent: true,
+                  },
+                  {
+                    begin: /#'\s?/,
+                    end: /$/,
+                    excludeBegin: true,
+                  }
+                ]
+              }
+            },
+            {
+              // Handle `@param` to highlight the parameter name following
+              // after.
+              className: 'doctag',
+              begin: '@param',
+              starts: {
+                end: /$/,
+                contains: [
+                  {
+                    className: 'meta',
+                    begin: IDENT_RE,
+                    endsParent: true
+                  }
+                ]
+              }
+            },
+            {
+              className: 'doctag',
+              begin: /@[a-zA-Z]+/
+            },
+            {
+              className: 'meta-keyword',
+              begin: /\\[a-zA-Z]+/,
+            }
```

Comment:
What is this trying to accomplish?

Reply:
Nevermind I didn't see the `\\`? :-)  With all the escaping easy to miss.

Reply:
Roxygen documentation has tags (`@tagname`) and pseudo-LaTeX formatting instructions (`\code{foo}`). I used `meta-keyword` for the latter — is that not appropriate?

Here’s an example snippet:

```r
#' @param foo is an \code{integer}
```

The intended (and actual) parse is as follows:

```xml
<comment>
  #'
  <doctag>@param</doctag>
  <meta>foo</meta>
  is an
  <meta-keyword>\code</meta-keyword>
  {integer}
<comment>
```

Reply:
Yeah, meta-keyword is appropriate there I think.  I just didn't see the `\\` so eht whole rule made no sense to me. :)